modules/luci-mod-status/htdocs/luci-static/resources/view/status/syslog.js

```diff
@@ -7,12 +7,11 @@
 return view.extend({
 	retrieveLog: async function() {
 		return Promise.all([
-			L.resolveDefault(fs.stat('/sbin/logread'), null),
-			L.resolveDefault(fs.stat('/usr/sbin/logread'), null)
+			L.resolveDefault(fs.stat('/usr/libexec/syslog-wrapper'), null)
 		]).then(function(stat) {
-			var logger = stat[0] ? stat[0].path : stat[1] ? stat[1].path : null;
+			var logger = stat[0].path;
 
-			return fs.exec_direct(logger, [ '-e', '^' ]).then(logdata => {
+			return fs.exec_direct(logger).then(logdata => {
```

Comment:
It looks like you excised the ternary operator above - which more than anything looks like a sanity check in case `stat[n]` is null. Maybe a null check and handling isn't bad here after all. We would already be in some weird place because to arrive at this state implies that the wrapper is absent for some reason. I guess it's no worse anyway, because the old 'resolves' had null as a default. What does `exec_direct` do in that case?

Reply:
I could also bring back the ternary operator
` var logger = stat[0] ? stat[0].path : null;`,
but as you already said it would be a weird place and the Promise already returns null in that case.

**From the [docs](https://openwrt.github.io/luci/jsapi/LuCI.fs.html)**:
`exec_direct` executes the command while bypassing ubus with cgi-io helper applet at `cgi-bin/cgi-exec'. It is useful to fetch large command outputs, which may exceed the ubus message size limits. It also enforces the same access permission rules as the ubus base exec call.